src/factory.js

```diff
@@ -131,6 +132,12 @@ module.exports = ({ Component, createElement, PropTypes }) => {
                       completeStateKeys,
                       stateAccessor
                     )
+                    Object.defineProperty(stateProxy, k, {
+                      enumerable: true,
+                      get () {
+                        throw new Error('cannot access computed from itself')
```

Comment:
You should probably add the name of the computed in the message to help the user.